### PR TITLE
COOK-014 Bugfix: Steps are now sorted on the server [GP, RT]

### LINKED ISSUE #73: COOK_014 Bug: Sorting steps in the back end
The steps have to be sorted on the back end

### PR DESCRIPTION
**Pull request by tobiasrieger**
Co-authored-by: Peter Grubelnik <44753443+pegru@users.noreply.github.com>

Closes #73

### CODE REVIEW

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 1]
```diff
@@ -41,11 +49,16 @@ public RecipeController(RecipeRepository repository, StepRepository stepReposito
...
     Recipe getById(@PathVariable Long id) {
-        return repository.findById(id).orElseThrow(() -> new RecipeNotFoundException(id));
+        Recipe recipe = repository.findById(id).orElseThrow(() -> new RecipeNotFoundException(id));
+        return recipe;
     }
 
     @PostMapping("")
     public Recipe insert(@RequestBody Recipe recipe) {
+
+        if(recipe.getSteps() != null)
```
Comment (prdngr) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java:59]:
```suggestion
        if (recipe.getSteps() != null)
```

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 2]
```diff
@@ -61,6 +74,10 @@ Recipe update(@RequestBody Recipe newRecipe, @PathVariable Long id) {
             recipe.setIngredients(newRecipe.getIngredients());
             recipe.setIsFavorite(newRecipe.getIsFavorite());
             stepRepository.deleteInBatch(recipe.getSteps());
+
+            if(recipe.getSteps() != null)
```
Comment (prdngr) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java:78]:
```suggestion
            if (recipe.getSteps() != null)
```

**Review by baumannc1999 — APPROVED** [review]
-> good work, bug is now fixed
-> clear code simple to understand

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 3]
```diff
@@ -41,11 +49,16 @@ public RecipeController(RecipeRepository repository, StepRepository stepReposito
 
     @GetMapping("/{id}")
     Recipe getById(@PathVariable Long id) {
-        return repository.findById(id).orElseThrow(() -> new RecipeNotFoundException(id));
+        Recipe recipe = repository.findById(id).orElseThrow(() -> new RecipeNotFoundException(id));
+        return recipe;
```
Comment (prdngr) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java:53]: Is this change necessary?

**Review by pucksME — APPROVED** [review]
Looks good, i think the front-end will work as expected now!
